test/integration/tag_test.rb

```diff
@@ -42,4 +42,39 @@ def render(*)
       assert_equal(buf.object_id, output.object_id)
     end
   end
+
+  def test_tags_can_be_overwritten_using_parse_context
+    static_tag = Class.new(Tag) do
+      def render(*)
+        'static_tag'
+      end
+    end
+
+    original_tag = Class.new(Tag) do
+      def render(*)
+        'original_tag'
+      end
+    end
+
+    new_tag = Class.new(Tag) do
+      def render(*)
+        'new_tag'
+      end
+    end
+
+    tags_overwrite = Liquid::Template::TagRegistry.new
+    tags_overwrite['dynamic_tag'] = new_tag
+
+    with_custom_tag('static_tag', static_tag) do
+      with_custom_tag('dynamic_tag', original_tag) do
+        liquid = '{% static_tag %} {% dynamic_tag %}'
+
+        template = Liquid::Template.parse(liquid)
+        assert_equal('static_tag original_tag', template.render)
+
+        template_with_overwrite = Liquid::Template.parse(liquid, tags: tags_overwrite)
```

Comment:
It's subjective, but `tags` felt intuitive to me. In fact, the first thing I tried was passing `tags` to `parse_context` before checking whether the functionality existed or not.

Your interpretation of the PR description sounds correct. The main reason why we can't overwrite all tags is because core liquid tags use the same registry as custom tags. If this overwrote all custom tags, it would solve the problem.

I originally ran down the path of having two separate registries—one for core tags and one for custom tags, but the implementation started growing complex and this was a simpler solution.

```
the global namespace could still unintentionally inherit tags
```

This is a good point since templates are user input. We wouldn't want a false positive signal that a tag was parsed when there may be no rendering support. Having two separate registries for core and custom might be necessary.